test/unit/ObjectMakrTests/processFunctions.js

```diff
@@ -0,0 +1,35 @@
+define(["mocks"], function (mocks) {
+    return function () {
+        var expect = require("chai").expect;
+
+        it("processes the correct number of functions", function () {
+            // Arrange
+            var objectMaker = mocks.mockObjectMakr();
+
+            var inheritance = mocks.mockInheritance();
+
+            // Act
+            objectMaker.functions = {};
```

Comment:
I felt that because `processFunctions` does not override `.functions` (in the sense that that `keys.length` value would be 3 + however many functions were created off the constructor call) that it would be better to have the key count represent what was passed into `processFunctions`